osquery/tables/system/hash.cpp

```diff
@@ -139,7 +158,108 @@ std::string hashFromFile(HashType hash_type, const std::string& path) {
   }
 }
 
-namespace tables {
+/*
+ * FileHashCache implements persistent in-memory caching for files' hashes.
+ * this cache has LRU eviction policy. the hash is recalculated
+ * every time the mtime or size of the file changes.
+ */
+
+#include <errno.h>
```

Comment:
Keep all of your includes together, so bring these up to the top.

Reply:
done